test/utils/session_utils_test.rb

```diff
@@ -0,0 +1,143 @@
+        error = assert_raises(ShopifyAPI::Errors::MissingJwtTokenError) do
+          ShopifyAPI::Utils::SessionUtils.session_id_from_shopify_id_token(id_token: nil, online: true)
+        end
+
+        assert_equal("Missing Shopify ID Token", error.message)
+      end
+
+      def test_non_embedded_app_current_session_id_raises_cookie_not_found_error
+        ShopifyAPI::Context.stubs(:embedded?).returns(false)
+
+        [
+          nil,
+          {},
+          { "not-session-cookie-name": "not-this-cookie" },
+        ].each do |cookies|
+          error = assert_raises(ShopifyAPI::Errors::CookieNotFoundError) do
+            ShopifyAPI::Utils::SessionUtils.current_session_id(nil, cookies, true)
+          end
+          assert_equal("Session cookie not found for app", error.message)
+        end
+      end
+
+      def test_non_embedded_app_current_session_id_returns_id_from_cookie
+        ShopifyAPI::Context.stubs(:embedded?).returns(false)
+        expected_session_id = "cookie_value"
+        cookies = { ShopifyAPI::Auth::Oauth::SessionCookie::SESSION_COOKIE_NAME => expected_session_id }
+
+        assert_equal(
+          expected_session_id,
+          ShopifyAPI::Utils::SessionUtils.current_session_id(nil, cookies, true),
+        )
+      end
+
+      def test_embedded_app_current_session_id_raises_cookie_not_found_error
+        ShopifyAPI::Context.stubs(:embedded?).returns(true)
+
+        [
+          nil,
+          {},
+          { "not-session-cookie-name": "not-this-cookie" },
+        ].each do |cookies|
+          error = assert_raises(ShopifyAPI::Errors::CookieNotFoundError) do
+            ShopifyAPI::Utils::SessionUtils.current_session_id(nil, cookies, true)
+          end
+          assert_equal("JWT token or Session cookie not found for app", error.message)
+        end
+      end
+
+      def test_embedded_app_current_session_id_raises_missing_jwt_token_error
+        ShopifyAPI::Context.stubs(:embedded?).returns(true)
+
+        error = assert_raises(ShopifyAPI::Errors::MissingJwtTokenError) do
+          ShopifyAPI::Utils::SessionUtils.current_session_id("", nil, true)
+        end
+
+        assert_equal("Missing Bearer token in authorization header", error.message)
+      end
+
+      def test_embedded_app_current_session_id_returns_online_id_from_auth_header
+        ShopifyAPI::Context.stubs(:embedded?).returns(true)
+        expected_session_id = "#{@shop}_#{@user_id}"
+
+        assert_equal(
+          expected_session_id,
+          ShopifyAPI::Utils::SessionUtils.current_session_id(@auth_header, nil, true),
+        )
+      end
+
+      def test_embedded_app_current_session_id_returns_offline_id_from_auth_header
+        ShopifyAPI::Context.stubs(:embedded?).returns(true)
+        expected_session_id = "offline_#{@shop}"
+
+        assert_equal(
+          expected_session_id,
+          ShopifyAPI::Utils::SessionUtils.current_session_id(@auth_header, nil, false),
```

Comment:
I know it's not new code we added in this PR, but can we expect to raise the missing JWT error when a valid token is present in the header but doesn't follow the `Bearer {token}` format?